config/base/src/lib.rs

```diff
@@ -256,3 +235,103 @@ pub mod view {
+    pub trait Configuration: Serialize + DeserializeOwned {
+        /// Error type returned by methods of this trait
+        type Error;
+
+        /// Return the JSON value of a given field
+        /// # Errors
+        /// Fails if field was unknown
+        fn get(&self, field: &'_ str) -> Result<Value, Self::Error> {
+            self.get_recursive([field])
+        }
+
+        /// Return the JSON value of a given inner field of arbitrary inner depth
+        /// # Errors
+        /// Fails if field was unknown
+        fn get_recursive<'tl, T>(&self, inner_field: T) -> Result<Value, Self::Error>
+        where
+            T: AsRef<[&'tl str]> + Send + 'tl;
+
+        /// Get documentation of a given inner field of arbitrary depth
+        ///
+        /// # Errors
+        /// Fails if field was unknown
+        fn get_doc_recursive<'tl>(
+            field: impl AsRef<[&'tl str]>,
+        ) -> Result<Option<String>, Self::Error>;
+
+        /// Get documentation of a given field
+        /// # Errors
+        /// Fails if field was unknown
+        fn get_doc(field: &str) -> Result<Option<String>, Self::Error> {
+            Self::get_doc_recursive([field])
+        }
+
+        /// Return documentation for all fields in a form of a JSON object
+        fn get_docs() -> Value;
+
+        /// Get inner documentation for non-leaf fields
+        fn get_inner_docs() -> String;
+    }
+
+    /// Trait for configuration loading and deserialization
+    pub trait Configurable:
+        Serialize + DeserializeOwned + Sized + LoadFromEnv + LoadFromDisk
+    {
+        /// Error type returned by methods of this trait
+        type Error;
+
+        /// If any of the fields in `other` are filled, they
+        /// override the values of the fields in [`self`].
+        #[must_use]
+        fn combine(self, other: Self) -> Self;
+    }
+
+    /// Trait for configuration loading and deserialization from
+    /// the environment
+    pub trait LoadFromEnv {
+        /// Error type returned by methods of this trait
+        type Error;
+
+        /// Load configuration from the environment
+        ///
+        /// # Errors
+        /// - Fails if deserializing any field is unsuccessful.
+        fn load_environment(&mut self) -> Result<(), Self::Error>;
+    }
+
+    /// Trait for configuration loading and deserialization from disk
+    pub trait LoadFromDisk: Sized {
+        /// Error type returned by methods of this trait
+        type Error;
+
+        /// Construct [`Self`] from a path-like object.
+        ///
+        /// # Errors
+        /// - File not found.
+        /// - File found, but peer configuration parsing failed.
+        fn from_path<P: AsRef<Path> + Debug + Clone>(path: P) -> Result<Self, Self::Error>;
+    }
+
+    /// Trait for building the final config from a proxy one
+    pub trait Builder {
+        /// The resulting type (most likely [`Configuration`]).
```

Comment:
I'd change `most likely` to something else, e.g.:
```suggestion
        /// The resulting type ([`Configuration`] in most cases).
```